pystiche_papers/data/utils.py

```diff
@@ -39,3 +42,32 @@ def __iter__(self) -> Iterator[Tuple[int, ...]]:
 
     def __len__(self) -> int:
         return self.num_batches
+
+
+class DelayedExponentialLR(ExponentialLR):
+    r"""Decays the learning rate of each parameter group by gamma after the delay.
+
+    Args:
+        optimizer: Wrapped optimizer.
+        gamma: Multiplicative factor of learning rate decay.
+        delay: Number of epochs before the learning rate is reduced with each epoch.
+        **kwargs: Optional parameters for the
+             :class:`~torch.optim.lr_scheduler.ExponentialLR`.
+    """
+
+    last_epoch: int
+    gamma: float
+    base_lrs: List[float]
```

Comment:
I've recently learned that this annotates class attributes, but these are instance attributes. Thus, we should move this into `__init__` with `self.last_epoch: int` etc.